src/state/reducers/auth/index.js

```diff
@@ -24,149 +25,110 @@ import {
-    [AUTH_RESTORE_SESSION_FAIL]: state => ({
-      ...state,
-      restoring: false,
-      restoringError: true
-    }),
-    [AUTH_SET_PASSWORD_INIT]: state => ({ ...state, loading: true }),
-    [AUTH_SET_PASSWORD_SUCCESS]: state => ({
-      ...state,
-      loading: false,
-      error: null
-    }),
-    [AUTH_SET_PASSWORD_FAIL]: (state, payload) => ({
-      ...state,
-      loading: false,
-      error: payload.error
-    }),
-    [AUTH_RESET_PASSWORD_INIT]: () => ({
-      ...initialState,
-      loading: true
-    }),
-    [AUTH_RESET_PASSWORD_SUCCESS]: state => ({
-      ...state,
-      loading: false,
-      error: null,
-      restoredPassword: true
-    }),
-    [AUTH_RESET_PASSWORD_FAIL]: (state, payload) => ({
-      ...state,
-      loading: false,
-      error: payload.error
-    }),
-    [AUTH_CLEAN_UP]: state => ({
-      ...state,
-      error: null,
-      changedPassword: false
-    }),
-    [AUTH_CHANGE_PASSWORD_INIT]: state => ({
-      ...state,
-      loading: true
-    }),
-    [AUTH_CHANGE_PASSWORD_SUCCESS]: state => ({
-      ...state,
-      loading: false,
-      changedPassword: true
-    }),
-    [AUTH_CHANGE_PASSWORD_FAIL]: (state, payload) => ({
-      ...state,
-      loading: false,
-      error: payload.error
-    }),
-    [AUTH_UPDATE_USER_DATA]: (state, payload) => ({
-      ...state,
-      userData: {
-        id: payload.id,
-        email: state.userData.email,
-        isAdmin: payload.isAdmin,
-        name: payload.name,
-        location: payload.location,
-        logoUrl: payload.logoUrl || state.userData.logoUrl,
-        createdAt: payload.createdAt
-      }
-    }),
-    [AUTH_PROVIDER_INIT]: state => ({
-      ...state,
-      loading: true
-    }),
-    [AUTH_PROVIDER_SUCCESS]: (state, payload) => ({
-      ...state,
-      userData: {
-        id: payload.id,
-        isAdmin: payload.isAdmin,
-        email: payload.email,
-        name: payload.name,
-        location: payload.location,
-        logoUrl: payload.logoUrl,
-        createdAt: payload.createdAt
-      },
-      error: null,
-      loading: false
-    }),
-    [AUTH_PROVIDER_FAIL]: (state, payload) => ({
-      ...state,
-      loading: false,
-      error: payload.error
-    })
-  },
-  initialState
-);
+export const authReducer = createReducer(initialState, {
+  [AUTH_SIGN_IN_INIT]: (state) => {
+    state.loading = true;
+  },
+  [AUTH_SIGN_IN_FAIL]: (state, { payload }) => {
+    state.loading = false;
+    state.error = payload.error;
+  },
+  [AUTH_FETCH_USER_DATA_INIT]: (state) => {
+    state.loading = true;
+  },
+  [AUTH_FETCH_USER_DATA_SUCCESS]: (state, { payload }) => {
+    const { user } = payload;
+    state.userData = user || payload; // || payload to fix errors while testing
+    state.loading = false;
+    state.error = null;
+    
+  },
+  [AUTH_FETCH_USER_DATA_FAIL]: (state, { payload }) => {
+    state.loading = false;
+    state.error = payload.error;
+  },
+
+  [AUTH_LOGOUT_INIT]: (state) => state,
```

Comment:
I think it might be a better implementation to change this to have the loading in true